### PR TITLE
[Search] Reducing the number of managed indexes in order to increase performances

### PR DESCRIPTION
**Pull request by NicolasEYSSERIC**
Optimizing search facets to improve response time too

### CODE REVIEW

File: core-library/src/main/java/org/silverpeas/core/index/search/model/IndexSearcher.java [hunk 1]
```diff
@@ -458,9 +486,11 @@ private MatchingIndexEntry createMatchingIndexEntry(ScoreDoc scoreDoc, String re
     if (topDocs != null) {
       for (int i = 0; i < topDocs.scoreDocs.length; i++) {
         ScoreDoc scoreDoc = topDocs.scoreDocs[i];
-        MatchingIndexEntry indexEntry = createMatchingIndexEntry(scoreDoc, query
-            .getRequestedLanguage(), searcher);
-        results.add(indexEntry);
+        final MatchingIndexEntry indexEntry =
+            createMatchingIndexEntry(scoreDoc, query.getRequestedLanguage(), searcher);
+        if (indexEntry != null) {
```
Comment (silveruser) [inline core-library/src/main/java/org/silverpeas/core/index/search/model/IndexSearcher.java]: MAJOR Remove this expression which always evaluates to "true" [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS2589)

File: core-war/src/main/java/org/silverpeas/web/pdc/control/PdcSearchSessionController.java [hunk 2]
```diff
@@ -538,15 +544,19 @@ private void processFacetFiletype(Facet facet, GlobalSilverResult result) {
     }
   }
 
-  private void processFacetComponent(Facet facet, GlobalSilverResult result, List<String> blackList) {
+  private void processFacetComponent(Facet facet, GlobalSilverResult result, List<String> blackList,
```
Comment (silveruser) [inline core-war/src/main/java/org/silverpeas/web/pdc/control/PdcSearchSessionController.java]: CRITICAL Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS3776)

**Comment by silveruser** [comment]
SonarQube analysis reported 4 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [PdcSearchSessionController.java#L28](https://github.com/Silverpeas/Silverpeas-Core/blob/0a99f73b0fc03c412a0f24bb6fcaeb88044f9483/core-war/src/main/java/org/silverpeas/web/pdc/control/PdcSearchSessionController.java#L28): Remove this unused import 'org.silverpeas.core.admin.component.model.SilverpeasComponentInstance'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [Admin.java#L70](https://github.com/Silverpeas/Silverpeas-Core/blob/0a99f73b0fc03c412a0f24bb6fcaeb88044f9483/core-library/src/main/java/org/silverpeas/core/admin/service/Admin.java#L70): Remove this unused import 'org.silverpeas.core.index.indexing.IndexFileManager'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by SilverYoCha** [comment]
Merged.
Closed manually because of manual git operations performed during integration.

**Pull request closed without merging**